chain-ser/src/deser.rs

```diff
@@ -84,3 +94,75 @@ impl<const N: usize> Deserialize for [u8; N] {
         Ok(buf)
     }
 }
+
+impl SerializedSize for u8 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u8>()
+    }
+}
+
+impl SerializedSize for u16 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u16>()
+    }
+}
+
+impl SerializedSize for u32 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u32>()
+    }
+}
+
+impl SerializedSize for u64 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u64>()
+    }
+}
+
+impl SerializedSize for u128 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u128>()
+    }
+}
```

Comment:
I wonder if could simplify these by implementing

`impl<T: Copy> SerializedSize for T`

but maybe it is a bit hacky?

Reply:
Unfortunately it is not possible and will not correct, because for any type which will implement `Copy` trait will have `std::mem::size_of` implementation which is not correct.

_The types *const T, &T, Box<T>, Option<&T>, and Option<Box<T>> all have the same size. If T is Sized, all of those types have the same size as usize._

Reply:
Oh right, I forgot about reference types, good point. Probably something like `zerocopy::AsBytes` would be a better fit, but that would require a dependency, so yeah, ignore this.

Reply:
we could use macros though